docs/source/common/test_set.rst

```diff
@@ -1,17 +1,26 @@
 .. _test_set:
 
-Test set
-========
-Lightning forces the user to run the test set separately to make sure it isn't evaluated by mistake.
-Testing is performed using the ``trainer`` object's ``.test()`` method.
+##########
+Evaluation
+##########
+
+*******
+Testing
+*******
+
+Lightning allows the user to test their models with any compatible test dataloaders. This can be done before/after training
+and is completely agnostic to :meth:`~pytorch_lightning.trainer.trainer.Trainer.fit` call. The logic used here is defined under
+:meth:`~pytorch_lightning.core.lightning.LightningModule.validation_step`.
+
+Testing is performed using the ``Trainer`` object's ``.test()`` method.
 
 .. automethod:: pytorch_lightning.trainer.Trainer.test
     :noindex:
 
-----------
 
-Test after fit
---------------
+Test after Fit
```

Comment:
We should add a note that benchmarking for papers with the test should be done with only a single device due to the DistributedSampler.

Reply:
good point! will mention